git/foundations_git/commit_messages.md

```diff
@@ -20,14 +20,43 @@ Yes! Let me give you a quick list of reasons why:
 
 -   Having a good commit message history will also be helpful if you come back to a project you were working on after stepping away from it for a while. You likely won't remember your thought process and changes made when initially writing out the code. 
 
+### Bad vs. Good Commits
+
+When it comes to writing commits, it is crucial to know how to write them effectively. Here's an example of a bad commit message:
+
+`fix a bug`
+
+Even though it describes what you did, the message is too vague, which leaves the other developers on your team confused. A good commit message will have a body and a subject. Here is a template to keep in mind when creating your commit message.
+
+## Subject
+An abbreviation that specifies an action that your line of code is doing(e.g., feat and fix) 
+
+`Commit type:`
```

Comment:
Nit: Will learners understand what commit type means here? do we cover it elsewhere? IMO, we should leave the "commit type" out. This lesson common pretty early in the curriculum and knowing the difference between different types of commits might not be useful/practical for learners yet.

Reply:
Good point @KevinMulhern, but learners could benefit from learning the abbreviations early. It'll help them develop good commit writing habits now than having to relearn this later on when they work on open-source projects and at their jobs. This is something I experienced when I started learning Git, and I don't want them to go through the same exact thing.

Reply:
I appreciate that point of view, but if we include "commit type" we'll have to explain what that means. Which will require us to have full explanations of what feature, chore and fix commits are. Learners won't get any value from that at this, _very_ early point in the curriculum. It muddies the water and will distract them from the main points we're trying to get across with this lesson. 

I've no objections with adding it to a later Git lesson though.

Reply:
Hey @KevinMulhern. I thought about what you said and decided to go with your advice. By putting `fix` or `feat`, we'd have to explain what that is, and that could cause cognitive overload, something we want to avoid.

Reply:
haha it's all good, cognitive overload is a great way of putting it. We're big believers in drip feeding in things exactly when they are needed. Build learners knowledge up gradually without overwhelming them 💪